snowpack/src/commands/dev.ts

```diff
@@ -834,13 +835,16 @@ export async function startDevServer(commandOptions: CommandOptions): Promise<Sn
       }
 
       let code = wrappedResponse;
-      if (responseFileExt === '.js' && reqUrlHmrParam)
+      if (responseFileExt === '.js' && mtime)
         code = await transformEsmImports(code as string, (imp) => {
           const importUrl = path.posix.resolve(path.posix.dirname(reqPath), imp);
           const node = hmrEngine.getEntry(importUrl);
           if (node && node.needsReplacement) {
+            node.mtime = mtime;
             hmrEngine.markEntryForReplacement(node, false);
-            return `${imp}?${reqUrlHmrParam}`;
+          }
+          if (node && node.mtime) {
```

Comment:
so this is a tricky part of ESM HMR, since you don't actually control the module registry and can't swap modules in and out. To get around this limitation, modules behave differently based on whether `isHmrAccepted` is true or false:

- `true`: This module accepts HMR updates without triggering a full page reload. That means that the `accept` handler "accepts and applies" updates to the current module instance. New versions are loaded via `?mtime`, but they aren't replaced into are folded into the existing application via the accept handler. The original URL is the only one that exists in the application.
- `false`: This is a normal module. When it `needsReplacement`, a new version is loaded into the application. Multiple are loaded within an HMR-enabled session, but only the most recent one should be imported and referenced within the application.

**(tldr)** All of that context is just to say: this fix should only apply if `isHmrAccepted = false`. If `isHmrAccepted = true`, then the current behavior is expected.

Reply:
Okay, I don't totally understand the context, but I think I got the logic right. Can you give that a look when you get a chance, please? Thanks!